plugin/sourcemap.ts

```diff
@@ -25,8 +25,8 @@ const getBytesPerFileUsingSourceMap = (
     if (source != null) {
       const id = path.resolve(path.dirname(path.join(dir, bundleId)), source);
 
-      modules[id] = modules[id] || { id, renderedLength: 0 };
-      modules[id].renderedLength += 1;
+      modules[id] = modules[id] || { id, code: "" };
+      modules[id].code += code[i];
```

Comment:
There is no reason to try to combine code there. gzip and brotli sizes disabled for it. To get length need to get code point and get its length as surrogate pair